content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md

```diff
@@ -0,0 +1,807 @@
+
+If these steps are followed, we will end up with a Docker image. The steps are as follows.
+
+##### Define base image
+
+A base image is an image that is used to create all of your container images. Here we shall use Python as our base image.
+
+```docker
+FROM tiangolo/uvicorn-gunicorn-fastapi:python3.7
+```
+
+##### Create a working directory
+
+```docker
+WORKDIR /app
+```
+
+##### Copy the app into the new directory created
+
+```docker
+COPY ./app /app
+```
+
+##### Install in the new working directory
+
+```docker
+RUN pip install fastapi uvicorn
+```
+
+##### Expose the port to serve your application
+
+Docker will run on port `8000`.
+
+```docker
+EXPOSE 8000
+```
+
+##### Create an entry point to be used to execute our image
+
+```docker
+ENTRYPOINT ["uvicorn", "app:app --reload"]
+
+CMD ["uvicorn", "app.app:app", "8000"]
+```
+
+#### Create Docker image
+
+A Docker image contains application code, libraries, tools, dependencies, and other files needed to make an application run.
+
+```docker
+docker build -t fastapi-test-app:new .
+```
+
+##### The output is as shown
+
+This output shows the process used when creating a docker image.
+
+```bash
+Sending context building to the Docker daemon  34.90kb
+Step 1/7 : FROM tiangolo/uvicorn-gunicorn-fastapi:python3.7
+  --->db183g656y4h
+Step 2/7 : WORKDIR /app
+  --->Using Cache
+  --->5df25yffdpbc
+Step 3/7 : COPY ./app /app
+    --->Using Cache
+    --->25dffbfdjdf5
+Step 4/7 : RUN pip install fastapi uvicorn
+    --->Using Cache
+    --->edf81dffcdf5
+Step 5/7 : EXPOSE 8000
+    --->Using Cache
+    --->afd99eb62d2
+Step 6/7 : ENTRYPOINT ["uvicorn", "app:app --reload"]
+    --->07taebte2egd
+Removing intermediate container 4edte5ta382
+ ---> 2de6fstf5uv09
+step 7/7 : CMD ["uvicorn", "app.app:app", "8000"]
+Successfully built 2de6fstf5uv09
+Successfully tagged fastapi-test-app:new
+```
+
+##### Listing all of our created images
+
+To list all the docker images we had created earlier, you can use the following command.
+Our latest image is `fastapi-test-app`. This is the image we have just created with an id of `2de6fstf5uv09`.
+
+```docker
+docker image ls
+```
+
+- Output:
+
+```bash
+REPOSITORY                   TAG                 IMAGE ID            CREATED             SIZE
+fastapi-test-app             new                2de6fstf5uv09      3 minutes ago       1.34GB
+testing                      latest             d661f1t3e0b         2 weeks ago          994MB
+```
+
+#### Creating docker container
+
+Docker containers are the live, running instances of Docker images, users can interact with them, and administrators can adjust their settings and conditions using docker commands.
+
+```bash
+docker run -p 8000:8000 fastapi:new
+```
+
+Result:
+
+```bash
+e0f1bd4gv1f7t3dti5e89fd1o29341a50ete9hgad8ed0ye0ff27dt81667fu16b
+```
```

Comment:
Is this a private key or something you should not share with anyone? @lalith1403